Sources/NIOFileSystem/Internal/SystemFileHandle.swift

```diff
@@ -753,6 +747,12 @@ extension SystemFileHandle.SendableView {
                     relativeTo: .currentWorkingDirectory,
                     flags: materialization.exclusive ? [.exclusive] : []
                 )
+                #else
+                renameResult = Syscall.rename(
+                    from: createdPath,
+                    to: desiredPath,
+                    options: materialization.exclusive ? [.exclusive] : []
+                )
```

Comment:
Turns out this doesn't compile on Android, because `rename(from:to:options:)` is only defined on Darwin.

Reply:
Oops, yes, this variant is Darwin only as it has the `.exclusive` flag. I think we can avoid this by not supporting "atomic" file creation on Android when exclusive create is set (when it isn't set we can do a normal rename without the exclusive flag).